internal/provider/cached_image_resource.go

```diff
@@ -631,34 +632,56 @@ func extractEnvbuilderFromImage(ctx context.Context, imgRef, destPath string) er
 	return fmt.Errorf("extract envbuilder binary from image %q: %w", imgRef, os.ErrNotExist)
 }
 
-// NOTE: the String() method of Terraform values will evalue to `<null>` if unknown.
-// Check IsUnknown() first before calling String().
-type stringable interface {
-	IsUnknown() bool
-	IsNull() bool
-	String() string
+// The below interfaces are to handle converting Terraform types can be converted to a string.
+// The String() method on an attr.Value creates a 'human-readable' version of the type, which
+// leads to quotes, escaped characters, and other assorted sadness.
+type valuestringer interface {
+	ValueString() string
 }
 
-func appendKnownEnvToList(list types.List, key string, value stringable) types.List {
+type valuebooler interface {
+	ValueBool() bool
+}
+
+type valueint64er interface {
+	ValueInt64() int64
+}
+
+// tfValueToString converts an attr.Value to its string representation
+// based on its implementation of the above interfaces.
+func tfValueToString(val attr.Value) string {
+	if val.IsUnknown() || val.IsNull() {
+		return ""
+	}
+	if vs, ok := val.(valuestringer); ok {
```

Comment:
If we only need these once, then this form be fine:

```suggestion
	if vs, ok := val.(interface{ValueString() string}); ok {
```

Reply:
TIL you could do this!